addons/test_mail/tests/test_message_track.py

```diff
@@ -251,3 +252,30 @@ def cleanup():
             'description': 'Zizizatestmaildescription',
         })
         test_mail_record.with_context(default_parent_id=2147483647).write({'name': magic_code})
+
+    def test_track_invalid_selection(self):
+        # Test: Check that initial invalid selection values are allowed when tracking
+        old_convert = ofields.Selection.convert_to_cache
+
+        def _convert(self, value, record, validate=True):
+            # Disable validation for selection fields
+            return old_convert(self, value, record, False)
+
+        invalid_value = 'I love writing tests!'
+        with patch(ofields.__name__ + ".Selection.convert_to_cache", _convert):
```

Comment:
Well my point was to test if it's possible to track a change from a selection value that's not in the registry (an "invalid value"). But first in order to set it, I had to disable validation, as otherwise it would raise an error.

Reply:
Can't you update the selection values (removing one) after having entered some values instead of mocking the convert to cache ? I understand what you are trying to do but it seems we are patching the wrong method. If we try to simulate the fact that a value is not in the registry, can't we mock the selection in fields itself ?

Reply:
Or you can simply update the record directly in SQL. We don't care how we end up in this "invalid selection value in the database" situation.

We just test that we are not stuck in this state.

Reply:
Better indeed :D